inductiva/fluids/scenarios/wind_tunnel/post_processing.py

```diff
@@ -40,47 +39,99 @@ def __init__(self,
+    def get_object_data(self, time_step: int): # pylint: disable=unused-argument
+        """Get aerodynamics data over an object set inside the WindTunnel
+        
+        Current Support - OpenFOAM
 
-    # Initialize reader and define reading time-step
-    reader = pv.POpenFOAMReader(reading_file)
-    reader.set_active_time_value(time_step)
+        Args:
+            time_step: Time step for which we want to visualize
+                aerodynamics data.
+        """
 
-    mesh = reader.read()
-    object_data = mesh["boundary"]["object"]
+        reading_file = os.path.join(self.sim_output_dir, "foam.foam")
 
-    return object_data
+        # Create reading file
+        with open(reading_file, "w", encoding="utf-8") as file:
+            file.close()
 
+        # Initialize reader and define reading time-step
+        reader = pv.POpenFOAMReader(reading_file)
+        reader.set_active_time_value(time_step)
 
-@WindTunnelSimulationOutput.get_pressure_field.register
-def _(self, simulator: OpenFOAM): # pylint: disable=unused-argument
-    """Returns pressure field over mesh points at a certain time_step."""
+        mesh = reader.read()
+        object_data = mesh["boundary"]["object"]
 
-    pressure_field = Object(self.object_data, "p")
+        return object_data
 
-    return pressure_field
+    def get_pressure_field(self): # pylint: disable=unused-argument
+        """Get pressure field over mesh points for a certain time_step.
+        
+        Returns:
+            A MeshData object that allow to manipulate data over a mesh
+            and to render it.
+        """
+        pressure_field = MeshData(self.object_data, "p")
+
+        return pressure_field
+
+    def get_streamlines(self,
+                        time_step: int,
+                        physical_property: str = "pressure"):
+        """Get streamlines over the object we insert in the WindTunnel."""
+
+        streamlines_path = os.path.join(
+            self.sim_output_dir, "postProcessing",
+            "sets", "streamLines",
+            str(time_step))
+
+        if physical_property.lower() == "pressure":
+            streamlines_mesh = pv.read(
+            os.path.join(streamlines_path, "track0_p.vtk"))
+        elif physical_property.lower() == "velocity":
+            streamlines_mesh = pv.read(
+            os.path.join(streamlines_path, "track0_U.vtk"))
+        else:
+            raise ValueError(f"Physical property `{physical_property}` not available "
+                             "or supported")
+
+        return streamlines_mesh
+
+    def get_flow_plane(self,
+                        time_step: int,
+                        physical_property: str = "pressure"):
+        """Get flow properties in a plane of the domain in WindTunnel."""
+        cutting_plane_path = os.path.join(
+            self.sim_output_dir, "postProcessing",
+            "cuttingPlane", str(time_step))
+
+        if physical_property.lower() == "pressure":
```

Comment:
I see a code repetition here, you could wrap this in a separate function.

Reply:
This was done right when you were reviewing :)